bundles/org.eclipse.e4.ui.workbench.swt/src/org/eclipse/e4/ui/workbench/swt/internal/copy/ViewPatternFilter.java

```diff
@@ -54,6 +54,10 @@ protected boolean isLeafMatch(Viewer viewer, Object element) {
 			if (wordMatches(text)) {
 				return true;
 			}
+			text = LocalizationHelper.getLocalized(desc.getCategory(), desc, context);
+			if (wordMatches(text)) {
+				return true;
+			}
```

Comment:
The entire second part of this method could be modernized and simplified:
```suggestion
		if (element instanceof MPartDescriptor desc) {
			for (String text : List.of(desc.getLabel(), desc.getCategory())) {
				if (wordMatches(LocalizationHelper.getLocalized(text, desc, context))) {
					return true;
				}
			}
		}
		return false;
	}
```
or you could even go one step further:
```suggestion
		return element instanceof MPartDescriptor desc //
				&& Stream.of(desc.getLabel(), desc.getCategory()).map(t -> LocalizationHelper.getLocalized(t, desc))
						.anyMatch(this::wordMatches);
	}
```

Reply:
Using Streams sounds like the best approach, given that this also makes the `if (element instanceof String)` redundant.